pkg/pillar/types/zedagenttypes.go

```diff
@@ -207,3 +207,33 @@ type DatastoreConfig struct {
 func (config DatastoreConfig) Key() string {
 	return config.UUID.String()
 }
+
+// NodeAgentStatus :
+type NodeAgentStatus struct {
+	Name              string
+	CurPart           string
+	TestInprogress    bool
+	UpdateInprogress  bool
+	RemainingTestTime time.Duration
+	NeedsReboot       bool
+	TestComplete      bool
+	RebootReason      string
+	ErrorStr          string
+}
+
+// Key :
+func (status NodeAgentStatus) Key() string {
+	return status.Name
+}
+
+// ZedAgentStatus :
+type ZedAgentStatus struct {
+	Name                   string
+	LastConfigReceivedTime time.Time
+	ConfigGetFailCount     int
```

Comment:
How is this useful? The last receive is what you want to look at, and that should suffice.

Reply:
removed.